server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RandomStoring.java

```diff
@@ -93,20 +91,21 @@ public void init() throws MessagingException {
 
     private Collection<ReroutingInfos> generateRandomMailboxes() {
         List<ReroutingInfos> reroutingInfos = this.reroutingInfos.block();
-        Collections.shuffle(reroutingInfos);
 
-        return reroutingInfos
-            .stream()
-            .limit(randomRecipientsNumbers.next())
+        return ThreadLocalRandom
```

Comment:
you should write a comment because I could really well see that code and say myself: "the author wanted to call Collections.shuffle()" and replace the code.